enterprise/glassfish.common/src/org/netbeans/modules/glassfish/common/GlassfishInstance.java

```diff
@@ -180,7 +182,7 @@ public boolean containsValue(Object value) {
 
         @Override
         public boolean containsKey(Object key) {
-            synchronized(delegate) {return delegate.containsKey(key);}
+            return delegate.computeIfPresent((String) key, (k, v) -> v) != null;
         }
```

Comment:
i don't understand why this changed.

Reply:
but a CHM does not require synchronization there. It simply answers the question: is the key in that map/set.

regular collections do have to be synchronized when using getters since there are no guarantees that a thread would ever see a result.

`containsKey` is perfectly fine here IMO.

(not to mention that you check for the value not being null, which is not always the same as checking if the key is present)

Reply:
I agree that just `containsKey` will suffice.

> (not to mention that you check for the value not being null, which is not always the same as checking if the key is present)

I checked the source code and find that the `containsKey` implementation of CHM is `get(key) != null` which checks for the value and the implementation for normal HM is `getNode(key) != null` which checks for the Node. This implies that the new code will work correctly for a CHM but not necessary for a HM, I have learned something new, thanks @mbien.